webpack.demo.js

```diff
@@ -1,39 +1,269 @@
+      name: ['vendor', 'polyfills']
+    }),
+
+    // Inject script and link tags into html files
+    // Reference: https://github.com/ampedandwired/html-webpack-plugin
     new HtmlWebpackPlugin({
-      template: './demo/index.html',
-      inject: 'body'
+      template: './demo/src/public/index.html',
+      inject: 'body',
+      chunksSortMode: packageSort(['polyfills', 'vendor', 'app'])
+    }),
+
+    // Extract css files
+    // Reference: https://github.com/webpack/extract-text-webpack-plugin
+    // Disabled when in test mode or not in build mode
+    new ExtractTextPlugin('css/[name].[hash].css', {disable: !isProd})
+  ];
+
+  // Add build specific plugins
+  if (isProd) {
+    config.plugins.push(
+      // Reference: http://webpack.github.io/docs/list-of-plugins.html#noerrorsplugin
+      // Only emit files when there are no errors
+      new webpack.NoErrorsPlugin(),
+
+      // Reference: http://webpack.github.io/docs/list-of-plugins.html#dedupeplugin
+      // Dedupe modules in the output
+      new webpack.optimize.DedupePlugin(),
+
+      // Reference: http://webpack.github.io/docs/list-of-plugins.html#uglifyjsplugin
+      // Minify all javascript, switch loaders to minimizing mode
+      new webpack.optimize.UglifyJsPlugin({
+        // Angular 2 is broken again, disabling mangle until beta 6 that should fix the thing
+        // Todo: remove this with beta 6
+        mangle: false
+      }),
+
+      // Copy assets from the public folder
+      // Reference: https://github.com/kevlened/copy-webpack-plugin
+      new CopyWebpackPlugin([{
+        from: root('demo/src/public')
+      }])
+    );
+  }
+
+  /**
+   * PostCSS
+   * Reference: https://github.com/postcss/autoprefixer-core
+   * Add vendor prefixes to your css
+   */
+  config.postcss = [
+    autoprefixer({
+      browsers: ['last 2 version']
     })
-  ],
-  devServer: {
-    contentBase: './demo/dist',
-    stats: {
-      modules: false,
-      cached: false,
-      colors: true,
-      chunk: false
+  ];
+
+  /**
+   * Sass
+   * Reference: https://github.com/jtangelder/sass-loader
+   * Transforms .scss files to .css
+   */
+  config.sassLoader = {
+    //includePaths: [path.resolve(__dirname, "node_modules/foundation-sites/scss")]
+  };
+
+  /**
+   * Apply the tslint loader as pre/postLoader
+   * Reference: https://github.com/wbuchwalter/tslint-loader
+   */
+  config.tslint = {
+    emitErrors: false,
+    failOnHint: false
+  };
+
+  /**
+   * Dev server configuration
+   * Reference: http://webpack.github.io/docs/configuration.html#devserver
+   * Reference: http://webpack.github.io/docs/webpack-dev-server.html
+   */
+  config.devServer = {
+    contentBase: './demo/src/public',
+    historyApiFallback: true,
+    stats: 'minimal' // none (or false), errors-only, minimal, normal (or true) and verbose
+  };
+
+  return config;
+}();
+
+// Helper functions
+function root(args) {
+  args = Array.prototype.slice.call(arguments, 0);
```

Comment:
No need for this, you can do `function root(...args) { ... }` - remember, webpack runs in Node, you have ES6!